parser/src/java/org/apache/hadoop/hive/ql/parse/HiveParser.g

```diff
@@ -1942,8 +1942,8 @@ metastoreCheck
 @after { popMsg(state); }
     : KW_MSCK (repair=KW_REPAIR)?
       (KW_TABLE tableName
-        (opt=(KW_ADD|KW_DROP|KW_SYNC) (parts=KW_PARTITIONS) (filterPartitionSpec)?)?)
-    -> ^(TOK_MSCK $repair? tableName? $opt? (filterPartitionSpec)?)
+        (opt=(KW_ADD|KW_DROP|KW_SYNC) (parts=KW_PARTITIONS)? (dropPartitionSpec)?)?)
```

Comment:
I think we are fighting with the complexity of the existing complexity :)

right now this will mean that we may write `TABLE t PARTITION PARTITION ...`  :D

do one of the following:
* either push out the `KW_PARTITION` keyword from `dropPartitionSpec` to every site it's being used
* or...remove the `(parts=KW_PARTITIONS)?` from here and change the `dropPartitionSpec` to accept it optionally by using: `(KW_PARTITION | KW_PARTITIONS)?` 

I think we should also clean up a bit:
* rename the `dropPartitionSpec` to something which doesn't have drop in its name like `partitionSelectorSpec`

Reply:
Option 2 looks good, Option 1 will change the current syntax.
Fixed.